cmd/dep/status_test.go

```diff
@@ -187,6 +187,13 @@ func TestBasicStatusGetConsolidatedConstraint(t *testing.T) {
 			},
 			wantConstraint: "1.2.1 (override)",
 		},
+		{
+			name: "BasicStatus with Revision Constraint",
+			basicStatus: BasicStatus{
+				Constraint: gps.Revision("ddeb6f5d27091ff291b16232e99076a64fb375b8"),
+			},
+			wantConstraint: "ddeb6f5",
+		},
```

Comment:
Thanks for adding this. But we need an integration test to test the actual change in result from this PR, because the `runStatusAll()` is not run by any of the unit tests. You can refer https://github.com/golang/dep/blob/master/cmd/dep/testdata/harness_tests/README.md to learn how to write and run the integration tests.

Reply:
Sounds good.  I'll work on the integration test